docs/language-spec.md

```diff
@@ -1487,79 +2108,123 @@ These different names refer to different ways of encoding the value on the wire.
+for closed enums (which could be a non-zero value) are not compatible with the
+semantics of optional fields with explicit presence.
 
-If the cardinality is omitted, `optional`, or `required`  then the field can have a
-normal, singular value. If the cardinality is `repeated` then the field can have one
-or more values and is represented as a list or an array of values.
+:::
 
-When the cardinality is `required`, a message is considered invalid if the field is
-not present.
+#### Cardinality and Field Presence
+
+Cardinality determines the number of values a field may have. Map fields and
+fields with `repeated` cardinality can have zero or more values, and are represented
+in generated code using a map (for the former) or a list or an array (for the latter).
+Other fields can have only a single value.
+
+In files that use proto2 or proto3 syntax, "field presence" is derived from the
+cardinality. In files that use the editions syntax, presence is configured via the
+[`field_presence` feature](#field-presence).
+
+Field presence indicates whether the field's presence or absence can be detected.
+* If a field's presence is required, it may never be absent.
+* If the field has explicit presence, its presence can always be detected.
+* If the field has implicit presence, its presence cannot be detected if the value
+  of the field is the default or zero value. Whether the field is present is
+  _implied by_ whether it has a default or zero value.
+
+The following table shows how field presence is derived from the field's cardinality:
+
+| Cardinality | Number of values          | Field Presence |
+|-------------|---------------------------|----------------|
+| `optional`  | 0 (absent) or 1 (present) | Explicit       |
+| `required`  | 1                         | Required       |
+| `repeated`  | 0 or more                 | Implicit       |
+| _map_ *     | 0 or more                 | Implicit       |
+| omitted     | _see below_               |                |
+
+__*__ [Map fields](#maps) behave like fields with `repeated` cardinality.
+      The main difference is that fields with `repeated` cardinality permit
+      duplicates and preserve element order. But map fields do not permit
+      entries that have duplicate keys and do not preserve element order.
+
+When the cardinality is omitted, the field presence depends on the context
+in which the field is defined:
+
+| File Syntax      | Context                                 | Field Presence                        |
+|------------------|-----------------------------------------|---------------------------------------|
+| _all_            | Inside a [oneof](#oneofs)               | Explicit                              |
+| proto3, editions | Inside an [`extend` block](#extensions) | Explicit                              |
+| proto3           | Normal field with message type          | Explicit                              |
```

Comment:
Does "normal" refer to a field without cardinality, that is not part of a oneof?

Reply:
Yes, good point. I'll define it. A "normal" field in this case is a non-extension field that is not in a oneof.

Reply:
Revised this to explain more clearly what is meant here.